src/test/integration/java/com/emt/service/MatchServiceIT.java

```diff
@@ -57,4 +64,74 @@ public void testEloRatingUpdateAfterMatch() {
     assertThat(updatedFirstPlayer.getEloRating()).isGreaterThan(initialRatingFirstPlayer);
     assertThat(updatedSecondPlayer.getEloRating()).isLessThan(initialRatingSecondPlayer);
   }
+
+  @Test
+  @Transactional
+  public void testCancelMatch_shouldRevertEloRatingsAndRemoveMatch() {
+    PlayerResponse winner =
+        playerService.createPlayer(CreatePlayerRequest.builder().nickname("Winner").build());
+    PlayerResponse loser =
+        playerService.createPlayer(CreatePlayerRequest.builder().nickname("Loser").build());
+
+    BigDecimal initialRatingWinner = winner.eloRating();
+
+    CreateMatchRequest matchRequest = new CreateMatchRequest(winner.playerId(), loser.playerId());
+    MatchResponse matchToCancel = matchService.createMatch(matchRequest);
+
+    matchService.cancelMatch(matchToCancel.matchId());
+
+    Player updatedWinner = playerService.getPlayerById(winner.playerId());
+
+    assertThat(updatedWinner.getEloRating().setScale(2, BigDecimal.ROUND_HALF_UP))
+        .isEqualTo(initialRatingWinner.setScale(2, BigDecimal.ROUND_HALF_UP));
+    assertThat(matchRepository.count()).isZero();
+  }
```

Comment:
This test is incorrect and incomplete
If you specify what you're checking in the method name (e.g. `AndRemoveMatch`), you should verify that the logic has done this correctly.
Also, you're canceling the match, so the rating should be `reverted`, but you assert that the `updatedWinner` rating is the same as the `initialRatingWinner`. This makes no sense